[MB-14914] Upgrade node.js to latest 18.x version (18.13.0)

reggieriser opened the pull request:
# Description

This PR updates our node version from 16.x to 18.13.0.  Support for node 16.x [will be ending in September 2023](https://nodejs.org/ar/blog/announcements/nodejs16-eol/).

To verify that nodejs 18.13.0 is in the container, do the following:

`docker run --rm milmove/circleci-docker:milmove-app-9ad7decf795181431073cae62dcc401d30558762 node --version`

The corresponding PR in the `mymove` repo that uses this is https://github.com/transcom/mymove/pull/9927.

## Changelog or Releases

[node 17.x changelog](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V17.md)
[node 18.x changelog](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md)

ahobson reviewed (approved):
🚀

duncan-truss reviewed (approved):
Tested out running the brew-upgrade and find-updates scripts, not sure I'd ever done that before.

I wonder if we really need the node version numbers in the scripts/brew-upgrade if it reads from `.node-version`?  It even sounds like `nodenv local` tries to update the `.node-version file`, which seems like it could end up reverting a version bump if they don't know to look in brew-upgrade?

reggieriser commented:
> I wonder if we really need the node version numbers in the scripts/brew-upgrade if it reads from `.node-version`? It even sounds like `nodenv local` tries to update the `.node-version file`, which seems like it could end up reverting a version bump if they don't know to look in brew-upgrade?

Good point.  I hadn't paid close attention to that and just changed all the node version numbers.  I removed the version numbers from both of those lines, so I believe the `nodenv install` should install whatever `.node-version` declares, and then the `nodenv local` will just report the local version.  That seemed how it played out when I tested it.
